### ISSUE TITLE
Unable to connect to Electrum Personal Server after upgrading to Electrum Wallet to 4.4.2

### ISSUE THREAD

**Issue by 1ddV** [spesmilo/electrum#8404]
Setup:
Electrum Wallet 4.4.2 running on Windows 11.
Bitcoin Core node 0.24.0.1 and Electrum Personal server 0.2.4 running on a Raspberry PI

Issue Description:
After the upgrade to to 4.4.2 release, my Electrum Wallet is unable to connect to my Electrum Personal Server (EPS 0.2.4 version).
The EPS logs continually shows the following connection & disconnection messages (I replaced the last IP octects with a "x"):
DEBUG:2023-05-07 17:04:55,546: Electrum connected from 192.168.x.x
DEBUG:2023-05-07 17:04:55,547: Electrum wallet disconnected
DEBUG:2023-05-07 17:05:05,825: Electrum connected from 192.168.x.x
DEBUG:2023-05-07 17:05:05,825: Electrum wallet disconnected

Actions attempted to fix the issue:
- Tried to enable the option to select servers automatically: the Electrum Wallet successfully connects to external servers. Then disabled the option and put my my <EPS IP>:50002 back. The problem didn't fix.
- Restarting both Electrum Wallet and Electrum Personal Server didn't fix the issue.

**Comment by SomberNight** [comment]
Anything interesting in the client's log?
https://electrum.readthedocs.io/en/latest/faq.html#how-to-enable-debug-logging

**Comment by 1ddV** [comment]
Thanks for the quick feedback.
Here it is:
20230507T162915.103759Z |     INFO | network | connecting to 192.168.x.x:50002:s as new interface
20230507T162917.143219Z |     INFO | interface.[192.168.x.x:50002] | disconnecting due to: ErrorGettingSSLCertFromServer(ConnectError(ConnectionRefusedError(22, 'The remote computer refused the network connection', None, 1225, None)))
20230507T162917.145184Z |     INFO | network | couldn't launch iface 192.168.x.x:50002:s -- CancelledError()
20230507T162925.840278Z |     INFO | network | connecting to 192.168.x.x:50002:s as new interface
20230507T162925.944671Z |     INFO | interface.[192.168.x.x:50002] | succeeded in getting cert
20230507T162926.167286Z |  WARNING | interface.[192.168.x.x:50002] | Cannot connect to main server due to SSL error (maybe cert changed compared to "C:\Users\xxxx\AppData\Roaming\Electrum\certs\192.168.x.x"). Exc: ConnectError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1129)'))
20230507T162926.167286Z |     INFO | network | couldn't launch iface 192.168.x.x:50002:s -- CancelledError()

**Comment by SomberNight** [comment]
> `20230507T162926.167286Z | WARNING | interface.[192.168.x.x:50002] | Cannot connect to main server due to SSL error (maybe cert changed compared to "C:\Users\xxxx\AppData\Roaming\Electrum\certs\192.168.x.x"). Exc: ConnectError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1129)'))`

Maybe your self-signed cert expired?

**Comment by 1ddV** [comment]
I don't remember generating a self signed certificate when configuring the Electrum Personal Server.
Anyway I agree it looks an issue with the validity of the SSL certificate...but where does Electrum Wallet get it?
I can close this issue and opening a new one to Electrum Personal Server? What do you suggest?

**Comment by 1ddV** [comment]
Maybe I can try this: https://github.com/spesmilo/electrum-server/blob/ce1b11d7f5f7a70a3b6cc7ec1d3e552436e54ffe/HOWTO.md#step-8-create-a-self-signed-ssl-cert

**Comment by SomberNight** [comment]
idk how EPS works, but maybe try finding where it stores the ssl cert it creates and delete that, it might generate a new one I guess.
Further, you can delete the cert cached and stored by the client at the path in the log message.

**Comment by 1ddV** [comment]
I deleted the cached cert stored by the client and nothing changed (it was recreated same as before).
I'll try to work on the EPS side tomorrow and update the issue.
Thanks for you help.

**Comment by 1ddV** [comment]
The issue is solved:
1) I regenerated the self-signed SSL cert on the raspberri using the procedure described at https://github.com/spesmilo/electrum-server/blob/ce1b11d7f5f7a70a3b6cc7ec1d3e552436e54ffe/HOWTO.md#step-8-create-a-self-signed-ssl-cert.
2) I had the EPS config.ini point to the new server.crt and server.key files.
3) On the windows client (Electrum Wallet) I deleted all files under %APPDATA%\Electrum\certs\

I'm closing the issue.
Thank you for your support @SomberNight
